packages/headless/src/app/search-engine/search-engine.ts

```diff
@@ -111,20 +112,26 @@ export interface SearchEngineOptions
 export function buildSearchEngine(options: SearchEngineOptions): SearchEngine {
   const logger = buildLogger(options.loggerOptions);
   validateConfiguration(options.configuration, logger);
-
   const searchAPIClient = createSearchAPIClient(options.configuration, logger);
-  const generatedAnswerClient = createGeneratedAnswerAPIClient(logger);
 
   const thunkArguments = {
     ...buildThunkExtraArguments(options.configuration, logger),
     apiClient: searchAPIClient,
-    streamingClient: generatedAnswerClient,
+    // The legacy client is used when the knowledgeConfigurationId is not provided
+    ...(options.configuration.answerConfigurationId
+      ? {}
+      : {streamingClient: createLegacyGeneratedAnswerAPIClient(logger)}),
   };
 
   const augmentedOptions: EngineOptions<SearchEngineReducers> = {
     ...options,
     reducers: searchEngineReducers,
     crossReducer: jwtReducer(logger),
+    // The middleware for the answer API is only added when the knowledgeConfigurationId is provided
+    ...(options.configuration.answerConfigurationId
+      ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        {middlewares: [answerAPI.middleware as any]} // TODO: types
```

Comment:
Help needed.
I couldn't find a way to comply here since the middlewares type is inferred from the SearchEngineReducers type and the latest can obviously not be modified.